Makefile

```diff
@@ -157,15 +156,31 @@ kind-deploy-controller: generate-operator-yaml install-resources deploy
 .PHONY: deploy-controller
 deploy-controller: kind-deploy-controller
 
+HOSTED ?= none
+
 .PHONY: kind-deploy-controller-dev
-kind-deploy-controller-dev: kind-deploy-controller build-images
+kind-deploy-controller-dev: 
+	if [ "$(HOSTED)" = "hosted" ]; then\
+		$(MAKE) kind-deploy-controller-dev-hosted ;\
+	else\
+		$(MAKE) kind-deploy-controller-dev-normal ;\
+	fi
+
+.PHONY: kind-deploy-controller-dev-normal
+kind-deploy-controller-dev-normal: kind-deploy-controller
 	@echo Pushing image to KinD cluster
 	kind load docker-image $(REGISTRY)/$(IMG):$(TAG) --name $(KIND_NAME)
 	@echo "Patch deployment image"
 	kubectl patch deployment $(IMG) -n $(KIND_NAMESPACE) -p "{\"spec\":{\"template\":{\"spec\":{\"containers\":[{\"name\":\"$(IMG)\",\"imagePullPolicy\":\"Never\"}]}}}}"
 	kubectl patch deployment $(IMG) -n $(KIND_NAMESPACE) -p "{\"spec\":{\"template\":{\"spec\":{\"containers\":[{\"name\":\"$(IMG)\",\"image\":\"$(REGISTRY)/$(IMG):$(TAG)\"}]}}}}"
 	kubectl rollout status -n $(KIND_NAMESPACE) deployment $(IMG) --timeout=180s
 
+.PHONY: kind-deploy-controller-dev-hosted
+kind-deploy-controller-dev-hosted:
```

Comment:
Eventually we'll want the regular non-hosted framework tests to also use the ManagedClusterAddon. In preparation for that, could we rename `kind-deploy-controller-dev-hosted` to something like `kind-deploy-controller-dev-addon`?

Reply:
Even though we're stripping the `-hosted` suffix (which I just realized), I think it should still be fine to use the same target.

Reply:
How about making "`-hosted`" dynamic (like a variable)? default "`hosted`"

Reply:
> How about making "`-hosted`" dynamic (like a variable)? default "`hosted`"

That'd work, but right now the `-hosted` string is only in one place and I can't think of another use case that it'd need to be overridden since other than hosted mode, we just want `KIND_NAMESPACE` as-is and the chances are low that it'd have a `-hosted` suffix in the name.